### PR TITLE
Address shoreline contamination class

### LINKED ISSUE #58: Develop workflow to incorporate shoreline contamination class

### PR DESCRIPTION
**Pull request by steeleb**
Heya Lindsay -

This in in part housekeeping, but then also me creating a vector file for our total AOI minus areas labeled as shoreline contamination.

Closes #58 - address shoreline contamination class

## Housekeeping:

1. Pulled the data folder local to this repo. This now houses the AOI files and will eventually house the labeler files, for posterity, too. This means that many of these files are related to AOI shapefiles and don't need review.
2. added a bunch of readme text to describe order of operations, give additional background info, etc. If there is anything that *isn't clear* please comment for more detail!
3. Moved some folders/files around a bit for clarity in the repo. This probably looks like trash from the GH side.

## Modeling progress

1. I generalized the collation code a smidge more for us (me, really) to run right before I start playing [deeply] with models.
2. I made some edits to the NHDPlusHR tiles to remove the harbor (from tile 1) and pain-in-the-butt bays from tiles 2 and 3. Those were done manually in QGIS. 
3. I created an additional AOI that has a 60m buffer around the 'shoreline contamination' class points from our dataset. 60m might be overkill, but it ends up doing a pretty good job of linking all of the points together so that I don't have to do anything extravagant to incorporate them. 

Something for you to think about:

When we are tallying up the pixel area that has been classified, how do you want me to account for those areas of contamination. Options are: a) don't include those areas in the summary - only include areas in the AOI that do not fall within the buffered shoreline contamination area, b) include them in the summary, and only flag the final rasters with possible shoreline contamination for the pixels that overlap those areas, c) include them as a separate summary column (i.e. total area minus the shoreline contamination for deep, near shore, etc AND a column that tallies the area for each class that is within that shoreline contamination area).

My gut says (a) or (c) - (b) would be part of both of those options.

### CODE REVIEW

File: data/aoi/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+## superior-plume-bloom/data/aoi
+
+This directory contains shapefiles used and created during various points within this workflow.
+
+The shapefile in the NHDPlusHR folder is the waterbody for HUC 0418 limited to FCode 390. This was obtained directly from the [National Map Download IDE](https://apps.nationalmap.gov/downloader/#/) and processed in QGIS.
+
+Files in the `tiledAOI` folder were created in the script `eePlumB/A_PrepAOI/01_TileAOI.Rmd`. Tiles 1-3 were edited manually in QGIS to remove the harbor from tile 1, and bays from tiles 2 and 3. These edited files have a suffix of `_noharbor` or `_no_bay`.
+
+The shapefiles `Superior_AOI_modeling` and `Superior_AOI_minus_shoreline_contamination` were created in the scripts `eePlumB/A_PrepAOI/02_Modeling_AOI.Rmd` and `modeling/01_Shoreline_Contamination.Rmd` respectively. The tiles were manually edited as described above between running these two scripts.
```
Comment (lindsayplatt) [inline data/aoi/README.md:9]: > The tiles were manually edited as described above between running these two scripts.

It is unclear to me what "as described above" is referring to. Is it that each of those shapefiles were opened in QGIS and the harbor/bay removed, then run through these two scripts? Or is the "above" the two .Rmd scripts listed?

**Review by lindsayplatt — COMMENTED** [review]
Just one small comment. 

As for what to do with shoreline contamination, my initial reaction is to choose option `c` where we keep shoreline contamination as a separate summary column. I am thinking it will be treated as if it were another label category. I think this could give us the most flexibility for using or not using it down the line.

**Review by lindsayplatt — APPROVED** [review]
Clarified!

**Comment by steeleb** [comment]
> Clarified!

Except that I was a derp and had it wrong. LOL. I don't know where my brain is. 


And I'm with you on choice (c). I think it's the most thorough and gives the most option down the road. I'm going to add one more file in here, then, for GEE - which is the union of those buffered shoreline contamination labels.

**Comment by steeleb** [comment]
Okay, should be all set! I'm going to merge this sucker. 

More to come. :)

**Pull request merged**